grr/server/grr_response_server/gui/ui/lib/api/api_interfaces.ts

```diff
@@ -36,6 +36,17 @@ export declare interface ApiClientLabel {
   readonly name?: string;
 }
 
+export declare interface ApiClientsLabels {
+  readonly items?: ApiClientLabel[];
+}
+/**
+ * AddClientsLabelsArgs proto mapping.
+ */
+export declare interface ApiAddClientsLabelsArgs {
+  readonly client_ids: string[];
```

Comment:
The API expects lowerCamelCase names. What is `client_ids` in the Protobuf should be called `clientIds` in TypeScript.

Reply:
Please use optional ReadonlyArray for all repeated Protobuf fields, e.g. `readonly clientIds?: ReadonlyArray<string>`. Generally, we try to keep data structures mostly immutable to leverage advantages of function programming (e.g. no unexpected modifications through side-effects, easy to understand code).

Reply:
Is there an exception to this rule?

Reply:
Yeah, we are not 100 % strict about the immutability. Sometimes the type system does not permit, e.g. NgRx expects a mutable `string[]`.